### PR TITLE
Improve timeout setting

### PR DESCRIPTION
**Pull request by bouwew**
While adding the config_flow updates in Core Plugwise, I realized we always start with a timeout of 30 secs, so testing a fail takes a long time. Then I thought, we know the password, only if equal to stretch set the timeout to 30 :)

## Summary by CodeRabbit

- **New Features**
	- Introduced improved timeout settings: 30 seconds for the Stretch device and 10 seconds for all other devices.

- **Bug Fixes**
	- Adjusted timeout assertions in tests to reflect new logic based on username conditions.

- **Refactor**
	- Simplified constructors by removing the timeout parameter across multiple classes.
	- Updated connection logic to eliminate timeout-related functionality.

- **Chores**
	- Updated project version to 1.4.2a0 in the configuration files.
	- Revised changelog to reflect the new version and improvements.

### CODE REVIEW

File: plugwise/helper.py [hunk 1]
```diff
@@ -74,9 +77,12 @@ def __init__(
         websession: ClientSession | None,
         username: str,
         port: int,
-        timeout: float,
     ) -> None:
         """Set the constructor for this class."""
+        timeout = DEFAULT_TIMEOUT
+        if username != DEFAULT_USERNAME:
+            timeout = DEFAULT_LEGACY_TIMEOUT
```
[the remark this answers is not in the document]
Comment (bouwew) [reply plugwise/helper.py:84]: @coderabbitai checking for invalid usernames is done in the frontend application.

File: tests/test_init.py [hunk 2]
```diff
@@ -552,7 +558,10 @@
                     await smile.full_update_device()
                     smile.get_all_devices()
                     data = await smile.async_update()
-                    assert smile._timeout == 30
+                    if smile.smile_type == pw_constants.STRETCH:
+                        assert smile._timeout == 30
+                    else:
+                        assert smile._timeout == 10
```
[the remark this answers is not in the document]
Comment (bouwew) [reply tests/test_init.py]: @coderabbitai this suggestion has been committed.

File: tests/test_init.py [hunk 3]
```diff
@@ -339,7 +339,7 @@ async def connect(
         )
 
         if not timeout:
-            assert smile._timeout == 30
+            assert smile._timeout == 10
```
[the remark this answers is not in the document]
Comment (bouwew) [reply tests/test_init.py:340]: @coderabbitai the `stretch` parameter is not relevant in de `connect` method.

**Review by CoMPaTech — APPROVED** [review]

**Comment by bouwew** [comment]
Closed, not implemented. A better solution has been implemented in the frontend Integration.

**Pull request closed without merging**